File: Huddle.MetricWebApp/Models/Metric.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Huddle.MetricWebApp.Models
{
    public class Metric:IQueryResult
    {
        public int Id { get; set; }

        public Issue Issue {get;set;}

        public string Name { get; set; }

        public string TargetGoal{ get; set; }

        public string ValueType { get; set; }
        public int State { get; set; }
        public DateTime StartDate { get; set; }
    }
}

File: Huddle.MetricWebApp/app/shared/models/metric.ts
import { Issue } from './issue';
import { State } from './state';
import { QueryResult } from './queryResult';

export class Metric implements QueryResult{
    id?: number;
    name?: string;
    issue?: Issue;
    targetGoal?: string;
    valueType?: string;
    metricState?: State;
    startDate?: Date;
}